protos/coinbase/chainstorage/blockchain_xrp.proto

```diff
@@ -0,0 +1,223 @@
+  string ledger_index = 2;       // The ledger index of the node.
+  message NewFields {
+    string exchange_rate = 1; // The exchange rate of the node.
+    string root_index = 2;      // The root index of the node.
+    string taker_gets_currency = 3; // The currency that the taker gets.
+    string taker_gets_issuer = 4;  // The issuer of the currency that the taker gets.
+    string taker_pays_currency = 5;  // The currency that the taker pays.
+    string taker_pays_issuer = 6;   // The issuer of the currency that the taker pays.
+  }
+  NewFields new_fields = 3; // The new fields of the node.
+}
+
+// TxResult represents the result of a transaction query.
+message LedgerTransaction {
+  string account = 1;            // The unique address of the account that initiated the transaction.
+  string fee = 2;                 // The amount of XRP, in drops, to be destroyed as a cost for distributing this transaction to the network.
+  uint32 flags = 3;                // The set of bit-flags for this transaction.
+  uint32 last_ledger_sequence = 4; // The highest ledger index this transaction can appear in.
+  uint32 offer_sequence = 5;      // The offer sequence number.
+  uint32 sequence = 6;            // The sequence number of the account sending the transaction.
+  string signing_pub_key = 7;     // The hex representation of the public key that corresponds to the private key used to sign this transaction.
+  TakerGets taker_gets = 8;        // The amount that the taker gets, specified in the offer.
+  string taker_pays = 9;         // The amount that the taker pays, specified in the offer.
+  string transaction_type = 10;   // The type of transaction, e.g., "OfferCreate".
+  string txn_signature = 11;      // The signature that verifies this transaction as originating from the account it says it is from.
+  string hash = 12;               // The hash of the transaction.
+  string ctid = 13;               // The currency transaction ID.
+  Meta meta = 14;                // The metadata associated with the transaction.
+  bool validated = 15;            // Indicates whether the transaction is validated.
+  int64 date = 16;                // The timestamp of the transaction, represented as the number of seconds since the Unix epoch.
+  uint32 ledger_index = 17;         // The index of the ledger that contains this transaction.
+  uint32 in_ledger = 18;          // The ledger index where this transaction was included.
+  string status = 19;             // The status of the transaction.
+  repeated Warning warnings = 20;   // An array of warning objects.
+}
+
+// Meta contains the metadata information related to a transaction.
+message Meta {
+  repeated AffectedNode affected_nodes = 1; // The list of nodes affected by the transaction.
+  uint32 transaction_index = 2;         // The index of the transaction within the ledger.
+  string transaction_result = 3;       // The result of the transaction.
```

Comment:
I'm not sure if the transaction_result needs to be modeled more than just string.  https://xrpl.org/docs/references/protocol/transactions/transaction-results/

Reply:
I am also not super sure about this part though, if we think we don't need it, I can change it to string.